lib/bugsnag.ex

```diff
@@ -61,6 +61,7 @@ defmodule Bugsnag do
     |> send_notification
     |> case do
       {:ok, %{status_code: 200}}   -> :ok
+      {:ok, %{reason: 'not_sent'}} -> {:ok, :not_sent}
```

Comment:
I'm not sure if `reason` is the right key name here. It's not super intention revealing. Also, if we're using a string as the value, perhaps something a bit richer than `not_sent` would be nice.

I can't think of something better than `reason` off the top of my head, and I realise we use it below for the `:error` tuple. I'll ponder for a bit but won't delay this unnecessarily.